magefiles/rulesengine/repos/infra_deployments.go

```diff
@@ -1,17 +1,152 @@
-		Description: "Run the default test suites which include the demo and components suites.",
-		Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
-
-			return rctx.RepoName == "infra-deployments"
-		}),
-		Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
-			rctx.LabelFilter = "e2e-demo,konflux-demo,spi-suite,remote-secret,integration-service,ec,build-templates,multi-platform"
-			return ExecuteTestAction(rctx)
-		})}},
+import (
+	"fmt"
+	"strings"
+
+	"github.com/konflux-ci/e2e-tests/magefiles/rulesengine"
+)
+
+// Default Rule of repo infra-deployments running konflux-demo suite.
+var InfraDeploymentsDefaultTestRules = rulesengine.Rule{Name: "Infra Deployments Default Test Execution",
+	Description: "Run the Konflux-demo suite tests when an Infra-deployments PR includes changes to files outside of the specified components.",
+	Condition: rulesengine.None{
+		&InfraDeploymentsIntegrationTestFileChangeRule,
+		&InfraDeploymentsImageControllerTestFileChangeRule,
+		&InfraDeploymentsMultiPlatformTestFileChangeRule,
+		&InfraDeploymentsBuildTemplatesTestFileChangeRule,
+		&InfraDeploymentsBuildServiceTestFileChangeRule,
+		&InfraDeploymentsReleaseServiceTestFileChangeRule,
+		&InfraDeploymentsEnterpriseControllerTestFileChangeRule,
+		&InfraDeploymentsJVMTestFileChangeRule,
+		rulesengine.ConditionFunc(CheckNoFilesChanged)},
+
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(ExecuteInfraDeploymentsDefaultTestAction)}}
+
+// ExecuteInfraDeploymentsDefaultTestAction excutes all the e2e-tests and component suites
+func ExecuteInfraDeploymentsDefaultTestAction(rctx *rulesengine.RuleCtx) error {
+	rctx.LabelFilter = "konflux"
+	return ExecuteTestAction(rctx)
+}
+
+// InfraDeploymentsTestFilesOnlyRule defines rules of test suites running of each changed component
+var InfraDeploymentsTestFilesOnlyRule = rulesengine.Rule{Name: "Infra-deployments PR Components File Diff Execution",
+	Description: "Runs specific tests of changed component by infra-deployments PR.",
+	Condition: rulesengine.Any{
+		&InfraDeploymentsIntegrationTestFileChangeRule,
+		&InfraDeploymentsImageControllerTestFileChangeRule,
+		&InfraDeploymentsMultiPlatformTestFileChangeRule,
+		&InfraDeploymentsBuildTemplatesTestFileChangeRule,
+		&InfraDeploymentsBuildServiceTestFileChangeRule,
+		&InfraDeploymentsReleaseServiceTestFileChangeRule,
+		&InfraDeploymentsEnterpriseControllerTestFileChangeRule,
+		&InfraDeploymentsJVMTestFileChangeRule},
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		// Adding "konflux" to the label filter when component is updated
+		AddLabelToLabelFilter(rctx, "konflux")
+		return nil
+
+	}),
+		rulesengine.ActionFunc(ExecuteTestAction)}}
+
+var InfraDeploymentsIntegrationTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Integration component File Change Rule",
```

Comment:
sorry for the late notice.. will it be good change the naming of the rules for `infra-deployment` repo .. for readability improvement.. rename `InfraDeploymentsIntegrationTestFileChangeRule` to something like `InfraDeploymentsIntegrationComponentChangeRule`

Reply:
I'm fine with this, 
@psturc wdyt?

Reply:
makes sense to me